src/calculation/DetailedResultsBuilder.ts

```diff
@@ -299,10 +319,12 @@ export function createEpisodePopulationValues(
               );
               if (observResult !== undefined) {
                 // push obs onto an existing populationResult
-                if (!observResult.observations) {
+                if (!observResult.observations && observation) {
                   observResult.observations = [];
                 }
-                observResult.observations.push(observation);
+                if (observation) {
+                  observResult.observations.push(observation);
+                }
```

Comment:
Potential simplification for this block. Avoids checking the truthiness of observation twice, but not necessary if you think your version is clearer :)
```
if (observation){
    if (!observResult.observations){
        observResult.observations = [];
    }
    observResult.observations.push(observation);
}
```